src/Object/Api/Mastodon/InstanceV2/Configuration.php

```diff
@@ -0,0 +1,47 @@
+<?php
+/**
+ * @copyright Copyright (C) 2010-2023, the Friendica project
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace Friendica\Object\Api\Mastodon\InstanceV2;
+
+use Friendica\BaseDataTransferObject;
+use Friendica\DI;
+
+/**
+ * Class Configuration
+ *
+ * @see https://docs.joinmastodon.org/entities/Instance/
+ */
+class Configuration extends BaseDataTransferObject
+{
+	/** @var StatusesConfig */
+	protected $statuses;
+	/** @var MediaAttachmentsConfig  */
+	protected $media_attachments;
+	/** @var int */
+	protected $image_size_limit;
+
+	public function __construct()
```

Comment:
So, `BaseDataTransferObject` are meant to be as simple as possible, to provide structured types with a few dependencies and business logic inside of them. This is why most of them have constructor parameter just matching their property list and simply setting the values accordingly.

Ideally, the Config values and the child structures should be retrieved in the Module and then passed down to the relevant data transfer object constructor.
```suggestion
	public function __construct(StatusesConfig $statuses, MediaAttachmentsConfig $media_attachments, int $image_size_limit)
```
Let me know if you'd like me to show you a full implementation of the V2 instance structure.

Reply:
OK I can reorder that. I don't understand what you mean when you state, "Let me know if you'd like me to show you a full implementation of the V2 instance structure."

Reply:
You’ve already come up with a full working implementation so I’d understand if you weren’t feeling like redoing it a different way. In that case I can supply the implementation myself.

Reply:
> You’ve already come up with a full working implementation so I’d understand if you weren’t feeling like redoing it a different way. In that case I can supply the implementation myself.

Oh! I appreciate that but I'm fine doing the reordering. This is my first time implementing something like this so it's a good exercise to learn how to do it the project preferred way.

Reply:
Should this really just be returning this as one giant hashtable instead of all these sub-objects? If all the heavy lifting is being done at the module level all these sub-classes seem to maybe be adding noise.

Reply:
No, we're trying to move away from arbitrary arrays in an effort to add some structure to the code. We've had countless issues with missing/unexpected keys in associative arrays so I'd rather we knew exactly what we're handling, and data transfer objects are an effort towards that end.

Reply:
@MrPetovan I've completed my attempt at the refactoring.

Reply:
This looks so much better to me, thank you!